src/Uno.UI/UI/Xaml/SpecializedResourceDictionary.cs

```diff
@@ -31,6 +31,7 @@ internal class SpecializedResourceDictionary
 		public readonly struct ResourceKey
 		{
 			public readonly string Key;
+			public readonly Type TypeKey;
 			public readonly bool IsType;
```

Comment:
We can certainly remove the `IsType` as a result now, as it is only used for type keys explicitly. That's remove a field to copy during by-value parameters.

Reply:
@jeromelaban Yeah sure. Will remove it. Do you want the HashCode to use `(TypeKey != null).GetHashCode()` (match current semantics), or `TypeKey.GetHashCode()` (probably better)?

Reply:
It's likely that `TypeKey?.GetHashCode() ?? 0` will be good enough.